### PR TITLE
Clean up send/recv tags

### PR DESCRIPTION
**Pull request by madsbk**
Now using a `dict` for the endpoint tags.

This PR is part **1** of an effort to merge https://github.com/rapidsai/ucx-py/pull/648 in a divide-and-conquer approach.

### CODE REVIEW

File: ucp/core.py [hunk 1]
```diff
@@ -472,28 +478,17 @@ class Endpoint:
...
-        self._ctrl_tag_send = ctrl_tag_send
-        self._ctrl_tag_recv = ctrl_tag_recv
         self._guarantee_msg_order = guarantee_msg_order
         self._send_count = 0  # Number of calls to self.send()
         self._recv_count = 0  # Number of calls to self.recv()
         self._finished_recv_count = 0  # Number of returned (finished) self.recv() calls
         self._shutting_down_peer = False  # Told peer to shutdown
         self._close_after_n_recv = None
+        self._tags = tags
+        assert tags is not None
```
Comment (pentschev) [inline ucp/core.py:491]: I don't think this line makes sense. Why have `tags=None` in the kwargs and assert here it isn't `None`?

**Review by pentschev — COMMENTED** [review]
I have a question here @madsbk , but otherwise I think it's a good idea to work on multiple PRs and unblock https://github.com/rapidsai/ucx-py/pull/648 . Thanks for working on this.

File: ucp/core.py (hunk 1 above)
Comment (madsbk) [reply ucp/core.py:491]: Oops, you are right that was only for debugging :)

**Review by pentschev — APPROVED** [review]
Thanks @madsbk !

**Comment by pentschev** [comment]
@gpucibot merge

**Pull request merged**